src/ansys/aedt/core/application/analysis_3d.py

```diff
@@ -214,20 +214,17 @@ def components3d(self):
         # libs = [syslib, userlib]
 
         libs = [
-            os.path.join(self.syslib, "3DComponents", self._design_type),
-            os.path.join(self.userlib, "3DComponents", self._design_type),
+            Path(self.syslib) / "3DComponents" / self._design_type,
+            Path(self.userlib) / "3DComponents" / self._design_type,
         ]
 
         for lib in libs:
-            if os.path.exists(lib):
+            if lib.exists():
                 listfiles = []
-                for root, _, files in os.walk(lib):
-                    for file in files:
-                        if file.endswith(".a3dcomp"):
-                            listfiles.append(os.path.join(root, file))
+                for file in lib.rglob("*.a3dcomp"):
+                    listfiles.append(file)
                 for el in listfiles:
-                    head, tail = ntpath.split(el)
-                    components_dict[tail[:-8]] = el
+                    components_dict[el.stem] = str(el)
```

Comment:
```suggestion
            if lib.exists():
                for file in lib.rglob("*.a3dcomp"):
                    components_dict[file.stem] = str(file)
```
I like the refactoring here but I think it can be reduced even more, no ?